src/consensus/dkg.rs

```diff
@@ -368,6 +357,117 @@ struct Participant {
     timer_token: u64,
 }
 
+impl Participant {
+    fn process_message(
+        &mut self,
+        node: &Node,
+        dkg_key: DkgKey,
+        message: DkgMessage,
+    ) -> Result<Vec<Command>> {
+        if self.dkg_key != dkg_key {
+            return Ok(vec![]);
+        }
+
+        let responses = self
+            .key_gen
+            .handle_message(&mut rand::thread_rng(), message)
+            .unwrap_or_default();
+
+        let mut commands = vec![];
+        for response in responses {
+            commands.extend(self.broadcast(node, dkg_key, response)?);
+        }
+
+        Ok(commands)
+    }
+
+    fn broadcast(
+        &mut self,
+        node: &Node,
+        dkg_key: DkgKey,
+        dkg_message: DkgMessage,
+    ) -> Result<Vec<Command>> {
+        let recipients: Vec<_> = self
+            .elders_info
```

Comment:
is this `self.elders_info` is the node's current `elders_info` or this dkg_voter's targeted elders_info?

Reply:
It's the proposed new elders info.